zmon-zauth-integration/src/main/java/org/zalando/zauth/zmon/service/ZauthAuthorityService.java

```diff
@@ -65,8 +68,11 @@ public ZauthAuthorityService(ZauthProperties zauthProperties, TeamService teamSe
         ZMonAuthority authority;
         if (groups.contains(zauthProperties.getAdminsGroup())) {
             authority = new ZMonAdminAuthority(username, ImmutableSet.copyOf(teamService.getTeams(username)));
-        } else {
+        } else if (groups.contains(zauthProperties.getUsersGroup())) {
```

Comment:
I am not really following. I though the problem is with calls to `getGroups`, which will be called with every service request, no?

Reply:
`getGroups` return ZACK group, so it's either `Apps/ZMON/Users` or `Apps/ZMON/Admins`. The problem was with calls to Teams API, i.e. `teamService.getTeams(username)`. Since call made by services will not return ZACK group, only access to dynamic teams service will be made (line 75).